src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -241,7 +271,9 @@ void GcsActorScheduler::HandleWorkerLeasedReply(
     // node, and then try again on the new node.
     RAY_CHECK(!retry_at_raylet_address.raylet_id().empty());
     auto spill_back_node_id = ClientID::FromBinary(retry_at_raylet_address.raylet_id());
-    if (auto spill_back_node = gcs_node_manager_.GetNode(spill_back_node_id)) {
+    auto spill_back_node = gcs_node_manager_.GetNode(spill_back_node_id);
+    if (spill_back_node != nullptr &&
+        !nodes_of_releasing_unused_workers_.contains(spill_back_node_id)) {
```

Comment:
I don't see where you add items to `nodes_of_releasing_unused_workers_`?

Reply:
Also, I think checking `nodes_of_releasing_unused_workers_` is not a good solution. Because:
1) It's easy to forget to check this in new code;
2) when a node is finished releasing, we need to trigger scheduling again.

I think maybe it's better to just ensure that `LeaseWorker` won't be sent before `RelaseUnsedWorker` is returned.

Reply:
fixed